src/actions/purchases/checkCoursePurchased.ts

```diff
@@ -0,0 +1,23 @@
+'use server';
+
+import { authOptions } from '@/lib/auth';
+import { getServerSession } from 'next-auth';
+import db from '@/db';
+
+export const checkCoursePurchased = async (courseId: string) => {
+  const session = await getServerSession(authOptions);
+  const user = session.user;
+
+  if (!user) {
+    return false;
+  }
+
+  const purchase = await db.userPurchases.findFirst({
```

Comment:
This wont work
Use `getPurchases` function instead
Sometimes purchases are not stored in the DB but still present in appx db